contributors/devel/api-conventions.md

```diff
@@ -620,18 +630,82 @@ containers:
 Delete operation will delete the first entry in the list that match the merge key.
 But there is validation to make sure no 2 entries with the same merge key will get in the list.
 
+Here is an example:
+
+Suppose we define a list of environment variables and we call them
+the original environment variables:
+```yaml
+env:
+  - name: ENV1
+    value: foo
+  - name: ENV2
+    value: bar
+  - name: ENV3
+    value: baz
+```
+
+Then the server appends another environment variable:
+```yaml
+env:
+  - name: ENV1
+    value: foo
+  - name: ENV2
+    value: bar
+  - name: ENV3
+    value: baz
+  - name: ENV4
+    value: server-added
+```
+
+Then the user wants to change it from the original to the following using `kubectl apply`:
+```yaml
+env:
+  - name: ENV1
+    value: foo
+  - name: ENV2
+    value: bar
+  - name: ENV5
+    value: new-env
+```
+
+The patch will looks like:
+```yaml
+$preserveOrderList/env:
+  - name: ENV1
+  - name: ENV2
+  - name: ENV5
+env:
+  - name: ENV3
+    $patch: delete
+  - name: ENV5
+    value: new-env
+```
+
+After server applying the patch:
+```yaml
+env:
+  - name: ENV1
+    value: foo
+  - name: ENV2
+    value: bar
+  - name: ENV5
+    value: new-env
+  - name: ENV4
+    value: server-added
+```
+
 #### List of Primitives
 
 We have two patch strategies for lists of primitives: replace and merge.
 Replace is the default patch strategy, which will replace the whole list on update and it will preserve the order;
-while merge strategy works as an unordered set. In this section, we call a primitive list with merge strategy an unordered set.
+while merge strategy works as an ordered set. In this section, we call a primitive list with merge strategy an ordered set.
 The patch strategy is defined in the go struct tag of the API objects,
 e.g. `finalizers` uses `merge` as patch strategy.
 ```go
 Finalizers []string `json:"finalizers,omitempty" patchStrategy:"merge" protobuf:"bytes,14,rep,name=finalizers"`
 ```
 
-##### Unordered Set
+##### Ordered Set
 
 There are 3 operations: add, delete, replace.
```

Comment:
Yeah, we shouldn't get rid of the current unordered set, we should add a way to deal with lists instead.